tpot/base.py

```diff
@@ -486,50 +486,65 @@ def pareto_eq(ind1, ind2):
+                                  'process. This could be because the data was '
+                                  'not formatted properly, or because data for '
+                                  'a regression problem was provided to the '
+                                  'TPOTClassifier object. Please make sure you '
+                                  'passed the data to TPOT correctly.')
+                        else:
+                            self._fitted_pipeline = self._toolbox.compile(expr=self._optimized_pipeline)
+
+                            with warnings.catch_warnings():
+                                warnings.simplefilter('ignore')
+                                self._fitted_pipeline.fit(features, classes)
 
-                    with warnings.catch_warnings():
-                        warnings.simplefilter('ignore')
-                        self._fitted_pipeline.fit(features, classes)
+                            if self.verbosity in [1, 2]:
+                                # Add an extra line of spacing if the progress bar was used
+                                if self.verbosity >= 2:
+                                    print('')
+                                print('Best pipeline: {}'.format(self._optimized_pipeline))
 
-                    if self.verbosity in [1, 2]:
-                        # Add an extra line of spacing if the progress bar was used
-                        if self.verbosity >= 2:
-                            print('')
-                        print('Best pipeline: {}'.format(self._optimized_pipeline))
+                            # Store and fit the entire Pareto front if sciencing
+                            elif self.verbosity >= 3 and self._pareto_front:
+                                self._pareto_front_fitted_pipelines = {}
 
-                    # Store and fit the entire Pareto front if sciencing
-                    elif self.verbosity >= 3 and self._pareto_front:
-                        self._pareto_front_fitted_pipelines = {}
+                                for pipeline in self._pareto_front.items:
+                                    self._pareto_front_fitted_pipelines[str(pipeline)] = self._toolbox.compile(expr=pipeline)
+                                    with warnings.catch_warnings():
+                                        warnings.simplefilter('ignore')
+                                        self._pareto_front_fitted_pipelines[str(pipeline)].fit(features, classes)
 
-                        for pipeline in self._pareto_front.items:
-                            self._pareto_front_fitted_pipelines[str(pipeline)] = self._toolbox.compile(expr=pipeline)
-                            with warnings.catch_warnings():
-                                warnings.simplefilter('ignore')
-                                self._pareto_front_fitted_pipelines[str(pipeline)].fit(features, classes)
+                    break
+                except:
```

Comment:
- To catch any exception use the ```except Exception```, instead of ```except BaseException``` or just ```except```.
- Indicate the most specific exception type in ```except``` block.

For example:
```python
# Bad
try:
    mapping[key]
except Exception:
    ...

# Better
try:
    mapping[key]
except KeyError:
    ...
```